homeassistant/components/homekit/accessories.py

```diff
@@ -85,7 +85,7 @@ def setup_char(char_name, service, value=None, properties=None, callback=None):
 
 
 def set_accessory_info(acc, name, model, manufacturer=MANUFACTURER,
-                       serial_number='0000'):
+                       serial_number=UNKNOWN):
```

Comment:
What do you think of removing the `manufacturer` parameter and just assigning the constant below.
The `serial_number` will always be given, so we don't need `UNKNOWN`.

Reply:
I think we should leave `manufacturer` for now because I wanted to look into allowing user-specified `manufacturer`, `model`, etc. using the HomeKit `entity_config`.

Reply:
We can leave it in there for now, but I don't think this should be done. To be honest I personally what prefer to only those required by HomeKit (`SerialNumber`) and the the ones necessary to identify the accessory (`entity_id` either in the `Model` or like your suggestion in the `SerialNumber`).

While writing this, is there anything useful you can do with this infos? Why not leave it to just `Manufacturer='Home Assistant'` and `SerialNumber=entity_id`?

Reply:
According to the Apple HomeKit Developer Guide, the Accessory Information service requires Manufacturer, Model, Name, Serial Number, and Firmware Revision so I think we should keep all of these characteristics for now.

![screen shot 2018-04-27 at 5 46 52 pm](https://user-images.githubusercontent.com/630673/39386619-849c08c8-4a43-11e8-8286-7e459e96a4d5.png)

I don't think there's much usefulness from allowing the user to specify the info, more of just a matter of making the user experience more beautiful/correct. My image is that it would be all optional config (i.e. Home Assistant = default manufacturer). I'll experiment with it, if it can be implemented cleanly and without a lot of config, then I'll submit a PR.

Reply:
You're right in that they are required chars. However everyone except for `Serial Number` can be empty strings. In the case of `Firmware Revision` the char would be hidden if the value is empty.

I will take a look at your PR (regarding the optional config) once it's finished. Have in mind however that every option you add needs to be maintained and therefore is extra work. Sometimes it's best to not do all thats possible.